jennchenn/uwexplore
# Allow users to reset password via reset link

## Implementation description

- added flask mail
- setup route so users can request to get a reset password link
- actual reset logic handled on Firebase side



## Steps to test

1. Rebuild containers (new package added) + add new environment variables to `.env` file
2. Send reset password request--verify response `{"success":true}`
```bash
curl -H "Content-Type: application/json" -d '{"email":"breadfydp+egg@gmail.com"}'  localhost:5000/auth/reset
```
3. Go to inbox and try to reset password by clicking on reset password link in email
4. Verify that new password is valid by using login route
```bash
curl -X POST -H "Content-Type: application/json" -d '{"email": "breadfydp+egg@gmail.com", "password": <new password>}' localhost:5000/auth/login
```
5. Try to reset password for non-existent user--verify no reset password link sent
```bash
curl -H "Content-Type: application/json" -d '{"email":"breadfydp+eggggg@gmail.com"}'  localhost:5000/auth/reset
```



## What should reviewers focus on?

- specific copy for email?
<img width="1116" alt="image" src="https://user-images.githubusercontent.com/32009013/224513314-917ea2ac-b404-4a6c-b251-3232b7caa76d.png">
